dot/sync/bootstrap_syncer_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_bootstrapSyncer_handleWorkerResult(t *testing.T) {
+	t.Parallel()
+
+	blockStateBuilderEmpty := func(ctrl *gomock.Controller) BlockState {
+		return NewMockBlockState(ctrl)
+	}
+
+	blockStateBuilder := func(ctrl *gomock.Controller) BlockState {
+		mockBlockState := NewMockBlockState(ctrl)
+		mockBlockState.EXPECT().BestBlockHeader().Return(&types.Header{Number: 2}, nil)
+		return mockBlockState
+	}
+
+	blockStateBuilderWithFinalised := func(ctrl *gomock.Controller) BlockState {
+		mockBlockState := NewMockBlockState(ctrl)
+		mockBlockState.EXPECT().BestBlockHeader().Return(&types.Header{Number: 2}, nil)
+		mockBlockState.EXPECT().GetHighestFinalisedHeader().Return(&types.Header{Number: 1}, nil)
+		return mockBlockState
+	}
```

Comment:
nit I would suggest inlining them in the test case field (unless they are repeated >= 3 times), to have things more 'local' which is easier when debugging a test case.

Reply:
Updated.